apps/dsp-app/src/app/workspace/resource/values/third-party-iiif/third-party-iiif.component.ts

```diff
@@ -0,0 +1,79 @@
+import { ChangeDetectorRef, Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormControl, Validators } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { iiifUrlValidator, infoJsonUrlValidatorAsync, previewImageUrlValidatorAsync } from './iiif-url-validator';
+import { IIIFUrl } from './third-party-iiif';
+
+@Component({
+  selector: 'app-third-part-iiif',
+  templateUrl: './third-party-iiif.component.html',
+  styleUrls: ['./third-party-iiif.component.scss'],
+})
+export class ThirdPartyIiifComponent implements OnInit, OnDestroy {
+  @Input() iiifUrlValue = '';
+  @Output() afterControlInit = new EventEmitter<FormControl<string>>();
+
+  iiifUrlControl: FormControl<string>;
+
+  previewImageUrl: string | null;
+  formStatus: 'VALIDATING' | 'LOADING' | 'IDLE' = 'IDLE';
+
+  private _destroy$ = new Subject<void>();
+
+  readonly validatorErrors = [
+    { errorKey: 'required', message: 'This field is required' },
+    { errorKey: 'invalidIiifUrl', message: 'The provided URL is not a valid IIIF image URL' },
+    { errorKey: 'previewImageError', message: 'The image cannot be loaded from the third party server' },
+    { errorKey: 'infoJsonError', message: 'The iiif info json cannot be loaded from the third party server' },
+  ];
+
+  readonly templateString = '{scheme}://{server}{/prefix}/{identifier}/{region}/{size}/{rotation}/{quality}.{format}';
+  readonly exampleString = 'https://example.org/image-service/abcd1234/full/max/0/default.jpg';
+
+  constructor(
+    private _cdr: ChangeDetectorRef,
+    private _fb: FormBuilder
+  ) {}
+
+  ngOnInit() {
+    this.iiifUrlControl = new FormControl(this.iiifUrlValue, {
+      validators: [Validators.required, iiifUrlValidator()],
+      asyncValidators: [previewImageUrlValidatorAsync(), infoJsonUrlValidatorAsync()],
+      updateOn: 'change',
```

Comment:
why is this updateOn: 'change' ?

Reply:
Oh, yes, thanks. This can be removed as well. (It triggers validation when the controls value changes, but it is the default behaviour of course). I introduced it once I faced issues in the workflow.